File: services/api/app/auth_models.py
"""Pydantic models for authentication requests and responses."""
from pydantic import BaseModel


class AppleSignInRequest(BaseModel):
    """Request model for Apple Sign In."""
    identity_token: str
    authorization_code: str


class AppleSignInResponse(BaseModel):
    """Response model for Apple Sign In."""
    auth_key: str

File: ios/FactCheck/FactCheck/AuthModels.swift
//
//  AuthModels.swift
//  FactCheck
//
//  Created on 2/6/26.
//

import Foundation

// MARK: - Request Models

struct AppleSignInRequest: Codable {
    let identity_token: String
    let authorization_code: String
}

// MARK: - Response Models

struct AppleSignInResponse: Codable {
    let auth_key: String
}
